mongo_connector/dest_mapping.py

```diff
@@ -0,0 +1,186 @@
+        self.reverse_plain = {}
+        # a dict containing plain db mappings, db -> an array of mapped db
+        self.plain_db = {}
+        
+        # the input namespace_set and ex_namespace_set could contain wildcard
+        self.namespace_set = namespace_set
+        self.ex_namespace_set = ex_namespace_set
+        
+        self.lock = threading.Lock()
+        
+        # initialize
+        for ns in namespace_set:
+            if ns not in inpt.keys():
+                inpt[ns] = ns
+
+        for k, v in inpt.items():
+            self.set(k,v)
+            
+    def match(self, src, dst_pattern):
+        """If source string src matches dst, return the matchobject"""
+        d_arr = re.split('(\*)',dst_pattern)
+        r_arr = []
+        for x in d_arr:
+            if x != "*":
+                r_arr.append(re.escape(x))
+            else:
+                r_arr.append("([^*]+)")
+                
+        reg_pattern = re.compile(''.join(r_arr))
+        m = reg_pattern.match(src)
+        return m
+    
+    def match_set(self, plain_src, dst_arr):
+        for x in dst_arr:
+            if plain_src == x:
+                return True
+            if "*" in x:
+                m = self.match(plain_src, x)
+                if m:
+                    return True
+        
+        return False
+    
+    def replace(self, src_match, map_pattern):
+        """Given the matchobject src_match, replace corresponding '*' in map_pattern with src_match."""
+        num_match = src_match.lastindex
+        if num_match != map_pattern.count("*"):
+            LOG.error("The number of wildcards in the mapping %s should be %d" %(map_pattern, num_match))
+            return None
+        pattern_arr = re.split('(\*)', map_pattern)
+        map_arr = []
+        map_index = 1
+        for x in pattern_arr:
+            if x != "*":
+                map_arr.append(x)
+            else:
+                map_arr.append(src_match.group(map_index))
+                map_index += 1
+        
+        return ''.join(map_arr)
+        
+    
+    def get(self,plain_src_ns,defaultval=""):
+        """Given a plain source namespace, return a mapped namespace if matched.
+        If no match and given defaultval, return it as a default value.
+        """
+        self.lock.acquire()
+        # search in plain mappings first
+        if plain_src_ns in self.plain.keys():
+            self.lock.release()
+            return self.plain[plain_src_ns]
+        else:
+            # search in wildcard mappings
+            # if matched, get a replaced mapped namespace
+            # and add to the plain mappings            
+            for k, v in self.wildcard.items():
+                m_res = self.match(plain_src_ns, k)
+                if m_res:
+                    res = self.replace(m_res, v)
+                    if res is not None:
+                        if res not in self.reverse_plain:
```

Comment:
Can you move this check (``if res not in self.reverse_plain:``) into the ``set_plain`` method? That would simplify this code a bit. 

Can you also add more information to the ``InvalidConfiguration`` error? Some useful info would be the actual values of the namespaces that clashed. Since this will halt mongo-connector we want to be as descriptive as possible.